ivy/functional/ivy/statistical.py

```diff
@@ -1148,6 +1148,98 @@ def cumprod(
     )
 
 
+@handle_array_function
+@to_native_arrays_and_back
+@handle_out_argument
+@handle_array_like_without_promotion
+@handle_nestable
+@handle_exceptions
+def cummax(
+    x: Union[ivy.Array, ivy.NativeArray],
+    axis: int = 0,
+    exclusive: bool = False,
+    reverse: bool = False,
+    *,
+    out: Optional[ivy.Array] = None,
+) -> ivy.Array:
+    """Return a namedtuple where values is the cumulative maximum of elements
+     of input in the given axis. And indices is the index location of each
+     maximum value found in the given axis.
+
+    Parameters
+    ----------
+    x
+        Input array.
+    axis
+        Axis along which the cumulative sum is computed. Default is ``0``.
+    exclusive
+        Whether to perform cummax exclusively. Default is ``False``.
+    reverse
+        Whether to perform the cummax from last to first element in the selected
+        axis. Default is ``False`` (from first to last element)
+    out
+        Optional output array, for writing the result to. It must have a shape that the
+        inputs broadcast to.
+
+    Returns
+    -------
+    ret
+        Array which holds the result of applying cummax at each
+        original array elements along the specified axis.
+
+    Examples
```

Comment:
Please fix the linting problems in the examples section, I have added some, please go through the rest :)

Reply:
Hi, I have fixed linting errors in the examples section. Let me know if you find any errors.